src/AspNetCoreModuleV2/CommonLib/hostfxr_utility.cpp

```diff
@@ -584,53 +562,37 @@ HOSTFXR_UTILITY::InvokeWhereToFindDotnet()
         fIsCurrentProcess64Bit = systemInfo.wProcessorArchitecture == PROCESSOR_ARCHITECTURE_AMD64;
     }
 
+    WLOG_INFOF(L"Current process bitness type detected as isX64=%d", fIsCurrentProcess64Bit);
+
     while (TRUE)
     {
         index = struDotnetLocationsString.IndexOf(L"\r\n", prevIndex);
         if (index == -1)
         {
             break;
         }
-        if (FAILED(hr = struDotnetSubstring.Copy(&struDotnetLocationsString.QueryStr()[prevIndex], index - prevIndex)))
-        {
-            goto Finished;
-        }
+
+        FINISHED_IF_FAILED(struDotnetSubstring.Copy(&struDotnetLocationsString.QueryStr()[prevIndex], index - prevIndex));
         // \r\n is two wchars, so add 2 here.
         prevIndex = index + 2;
 
-        if (GetBinaryTypeW(struDotnetSubstring.QueryStr(), &dwBinaryType) &&
-            fIsCurrentProcess64Bit == (dwBinaryType == SCS_64BIT_BINARY))
+        WLOG_INFOF(L"Processing entry %s", struDotnetSubstring.QueryStr());
+
+        if (LOG_LAST_ERROR_IF(!GetBinaryTypeW(struDotnetSubstring.QueryStr(), &dwBinaryType)))
         {
-            // The bitness of dotnet matched with the current worker process bitness.
-            result = std::make_optional(struDotnetSubstring.QueryStr());
-            fResult = TRUE;
-            break;
+            continue;
         }
-    }
 
-Finished:
+        WLOG_INFOF(L"Binary type %d", dwBinaryType);
 
-    if (hStdOutReadPipe != INVALID_HANDLE_VALUE)
-    {
-        CloseHandle(hStdOutReadPipe);
-    }
-    if (hStdOutWritePipe != INVALID_HANDLE_VALUE)
-    {
-        CloseHandle(hStdOutWritePipe);
-    }
-    if (processInformation.hProcess != INVALID_HANDLE_VALUE)
-    {
-        CloseHandle(processInformation.hProcess);
-    }
-    if (processInformation.hThread != INVALID_HANDLE_VALUE)
-    {
-        CloseHandle(processInformation.hThread);
-    }
-    if (pwzDotnetName != NULL)
-    {
-        SysFreeString(pwzDotnetName);
+        if (fIsCurrentProcess64Bit == (dwBinaryType == SCS_64BIT_BINARY))
+        {
+            // The bitness of dotnet matched with the current worker process bitness.
+            return std::make_optional(struDotnetSubstring.QueryStr());
+        }
     }
 
+    Finished:
```

Comment:
I don't like that we still need the goto. Macro that returns a value if the results failed?

Reply:
Nope, we need a THROW_IF* set of macros but that is a bit too much work. I want this PR out so the bug can be fixed first.

Reply:
Sure.